libs/components/modals/src/lib/modules/modal/modal.service.ts

```diff
@@ -65,7 +65,12 @@ export class SkyModalService {
     });
 
     if (SkyModalService.host) {
-      SkyModalService.host.instance.open(modalInstance, component, params);
+      SkyModalService.host.instance.open(
+        modalInstance,
+        component,
+        params,
+        this.#environmentInjector,
```

Comment:
By taking the environment injector from the service, the component will receive the injector that corresponds with the `providedIn` setting for either `SkyModalService` or `SkyModalLegacyService`.